src/components/Cards.tsx

```diff
@@ -1,129 +1,140 @@
-		setFlipCard(false);
-		setInputValue(prevInputValue);
-		setAnswerInputValue(prevAnswerInputValue);
-		setIsEditCard(false);
-		setShowError(false);
-	};
+  const submitChanges = () => {
+    if (inputValue && answerInputValue != "") {
+      setFlipCard(false);
+      setIsEditCard(false);
+      setShowError(false);
+    } else {
+      setShowError(true);
+    }
+  };
+  const cancelChanges = () => {
+    setFlipCard(false);
+    setInputValue(prevInputValue);
+    setAnswerInputValue(prevAnswerInputValue);
+    setIsEditCard(false);
+    setShowError(false);
+  };
 
-	return (
-		<>
-			{!isEditCard && (
-				<div
-					onClick={() => {
-						setFlipCard(!flipCard);
-					}}
-					className={`${styles.rectangle} ${flipCard ? styles.back : styles.front}`}>
-					<button
-						onClick={editCard}
-						className={styles.buttonTrash}>
-						<Pencil />
-					</button>
-					<div className={styles.text}>
-						<p className={styles.heading}>{inputValue}</p>
-						<p className={styles.answer}>{answerInputValue}</p>
-					</div>
-				</div>
-			)}
-			{isEditCard && (
-				<>
-					{flipCard ? (
-						<div className={styles.rectangle}>
-							<button
-								onClick={editCard}
-								className={styles.buttonTrash}>
-								<Trash />
-							</button>
-							<input
-								onChange={handleInputChange}
-								type='text'
-								value={inputValue}
-							/>
-							{showError && <p className={styles.errorParaph}>Wprowadź najpierw jakieś słowa.</p>}
-							<div className={styles.buttons}>
-								<button
-									onClick={cancelChanges}
-									className={styles.buttonCardOne}>
-									Cancel
-								</button>
-								<button
-									onClick={submitChanges}
-									className={styles.buttonCardSecond}>
-									Save
-								</button>
-							</div>
-						</div>
-					) : (
-						<div className={styles.rectangle}>
-							<button
-								onClick={editCard}
-								className={styles.buttonTrash}>
-								<Trash />
-							</button>
-							<p className={styles.headingTitle}>{inputValue}</p>
-							<input
-								onChange={handleAnswerInputChange}
-								type='text'
-								value={answerInputValue}
-							/>
-							{showError && <p className={styles.errorParaph}>Wprowadź najpierw jakieś słowa.</p>}
-							<div className={styles.buttons}>
-								<button
-									onClick={cancelChanges}
-									className={styles.buttonCardOne}>
-									Cancel
-								</button>
-								<button
-									onClick={submitChanges}
-									className={styles.buttonCardSecond}>
-									Save
-								</button>
-							</div>
-						</div>
-					)}
-				</>
-			)}
-		</>
-	);
+  return (
+    <>
+      {!isEditCard && (
+        <div
+          onClick={() => {
+            setFlipCard(!flipCard);
+          }}
+          className={`${styles.rectangle} ${
+            flipCard ? styles.back : styles.front
+          }`}
+        >
+          <button onClick={editCard} className={styles.buttonTrash}>
+            <Pencil />
```

Comment:
**nitpick:** You've got way too many components in this `Cards` component. It will be extremely difficult to maintain or debug this code. I would split it into eg: `CardFront`, and `CardBack` so you know exactly what is what. You will also have less code here after that.